src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DockerImageDownloader.java

```diff
@@ -310,15 +318,20 @@ private <T> T runWithConnectionErrorCheck(CrashableSupplier<T, Exception> task)
      */
     @Override
     public void cleanup() throws IOException {
-        // this docker image not only used by itself
         try {
             if (!ifImageUsedByOther(componentStore)) {
-                Image image = DockerImageArtifactParser
-                        .getImage(ComponentArtifact.builder().artifactUri(artifact.getArtifactUri()).build());
-                dockerClient.deleteImage(image);
+            RetryUtils.runWithRetry(finiteDeleteAttemptsRetryConfig, () -> {
+
+                    Image image = DockerImageArtifactParser
+                            .getImage(ComponentArtifact.builder().artifactUri(artifact.getArtifactUri()).build());
+                    dockerClient.deleteImage(image);
+                return null;
+            }, "docker-image-cleanup", logger);
             }
-        } catch (PackageLoadingException | InvalidArtifactUriException e) {
+        } catch (PackageLoadingException | InvalidArtifactUriException | InterruptedException e) {
             throw new IOException(e);
+        } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that your code handles a broad swath of exceptions in the catch block, potentially trapping dissimilar issues or problems that should not be dealt with at this point in the program.

[Learn more](https://cwe.mitre.org/data/definitions/396.html)